src/IronRure/Regex.cs

```diff
@@ -1,277 +1,293 @@
+
+    /// <summary>
+    ///   Capture All Matches
+    ///   <para>
+    ///     Returns an iterator containing the capture information
+    ///     for each match of the pattern.
+    ///   </para>
+    /// </summary>
+    /// <param name="haystack">The string to search for this pattern</param>
+    public IEnumerable<Captures> CaptureAll(byte[] haystack)
+    {
+        using CapturesIter iter = new(this, haystack);
+        while (iter.MoveNext())
         {
-            using (var iter =  new CapturesIter(this, haystack))
-            {
-                while (iter.MoveNext())
-                {
-                    yield return iter.Current;
-                }
-            }
+            yield return iter.Current;
         }
+    }
+
+    /// <summary>
+    ///   Capture All Matches
+    ///   <para>
+    ///     Returns an iterator containing the capture information
+    ///     for each match of the pattern.
+    ///   </para>
+    /// </summary>
+    /// <param name="haystack">The string to search for this pattern</param>
+    public IEnumerable<Captures> CaptureAll(string haystack)
+    {
+        byte[] haystackBytes = Encoding.UTF8.GetBytes(haystack);
+        return CaptureAll(haystackBytes);
+    }
 
-        /// <summary>
-        ///   Capture All Matches
-        ///   <para>
-        ///     Returns an iterator containing the capture information
-        ///     for each match of the pattern.
-        ///   </para>
-        /// </summary>
-        /// <param name="haystack">The string to search for this pattern</param>
-        public IEnumerable<Captures> CaptureAll(string haystack)
+    /// <summary>
+    ///   Releases the unmanaged resources used by the Regex and optionally releases the managed resources.
+    /// </summary>
+    /// <param name="disposing">
+    ///   true to release both managed and unmanaged resources; false to release only unmanaged resources.
+    /// </param>
+    protected virtual void Dispose(bool disposing)
+    {
+        if (_disposed)
         {
-            var haystackBytes = Encoding.UTF8.GetBytes(haystack);
-            return CaptureAll(haystackBytes);
+            return;
         }
 
-        /// <inheritdoc />
-        public void Dispose()
+        if (disposing)
         {
-            Raw.Dispose();
+            // Dispose managed resources if any
         }
 
-        /// <summary>
-        ///   The default flags for the regex
-        /// </summary>
-        public static RureFlags DefaultFlags => RureFlags.Unicode;
+        // Dispose unmanaged resources safely.
+        Raw?.Dispose();
+        _disposed = true;
+    }
+
+
+    /// <inheritdoc />
+    public void Dispose()
+    {
+        Dispose(true);
+        GC.SuppressFinalize(this);
+    }
+
+    /// <summary>
+    ///   The default flags for the regex
+    /// </summary>
+    public static RureFlags DefaultFlags => RureFlags.Unicode;
+
+    /// <summary>
+    ///   Finalizer
+    /// </summary>
+    ~Regex()
```

Comment:
The safe handle has the finaliser. I don't think we should make regex finalisable.

Reply:
That one should be sorted now.